jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1247,14 +1247,13 @@ protected KeyStore loadTrustStore(Resource resource) throws Exception
                 }
 
                 // Is SNI needed to select a certificate?
-                if (!_certWilds.isEmpty() || _certHosts.size() > 1 || (_certHosts.size() == 1 && _aliasX509.size() > 1))
+                boolean sniRequired = (this instanceof Server) && ((Server)this).isSniRequired();
+                if (sniRequired || !_certWilds.isEmpty() || _certHosts.size() > 1 || (_certHosts.size() == 1 && _aliasX509.size() > 1))
```

Comment:
Maybe a stupid question, but is it possible to move `newSniX509ExtendedKeyManager` wrapping to a `Server.getKeyManagers` method?

Reply:
@rnd4222 we have done this in Jetty 10, but in 9.4.x there are people still using `SslContextFactory` (not `SslContextFactory.Server`).